toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -366,12 +368,35 @@ func convertImageToRaw(ic *ImageCustomizerParameters) error {
 		sourceArg += ",driver=vpc,force_size_calc=current_size"
 	}
 
-	err = shell.ExecuteLiveWithErr(1, "qemu-img", "convert", "-O", "raw", "--image-opts", sourceArg, ic.rawImageFile)
+	err = shell.ExecuteLiveWithErr(1, "qemu-img", "convert", "-O", "raw", "--image-opts", sourceArg, rawImageFile)
 	if err != nil {
-		return fmt.Errorf("failed to convert image file to raw format:\n%w", err)
+		return "", fmt.Errorf("failed to convert image file to raw format:\n%w", err)
 	}
 
-	return nil
+	format, err := qemuStringtoImageFormatType(detectedImageFormat)
+	if err != nil {
+		return "", err
+	}
+	return format, nil
+}
+
+func qemuStringtoImageFormatType(qemuFormat string) (imagecustomizerapi.ImageFormatType, error) {
+	switch qemuFormat {
+	case "raw":
+		return imagecustomizerapi.ImageFormatTypeRaw, nil
+	case "qcow2":
+		return imagecustomizerapi.ImageFormatTypeQcow2, nil
+	case "vpc":
+		return imagecustomizerapi.ImageFormatTypeVhd, nil
+	case "vhdx":
+		return imagecustomizerapi.ImageFormatTypeVhdx, nil
+	case "iso":
+		return imagecustomizerapi.ImageFormatTypeIso, nil
+	case "cosi":
```

Comment:
I don't think `qemu-img` is capable of detecting COSI files or ISO files.

Reply:
I guess there will not be a case cosi to be the base and needs to be converted to raw? So I will just remove it here.

Reply:
We might want to support that in the future. But we might need to write our own detection logic for COSI files.